code/game/objects/structures/crates_lockers/crates.dm

```diff
@@ -525,6 +525,16 @@
 	else
 		to_chat(user, "<span class='notice'>Access Denied.</span>")
 
+/obj/structure/closet/crate/secure/proc/autotogglelock(var/obj/machinery/M)
```

Comment:
you could just tweak togglelock() to avoid this spaghetti

Reply:
Implemented in latest commit